src/hashutil.cpp

```diff
@@ -313,23 +313,27 @@ hash_command_output(struct hash* hash,
   }
 #endif
 
-  struct args* args = args_init_from_string(command);
-  for (int i = 0; i < args->argc; i++) {
+  Args args = Args::from_string(command);
+
+  for (size_t i = 0; i < args.size(); i++) {
     if (str_eq(args->argv[i], "%compiler%")) {
       args_set(args, i, compiler);
     }
   }
-  cc_log_argv("Executing compiler check command ", args->argv);
+
+  auto argv_keeper = args.to_argv();
+  auto argv = argv_keeper.get();
+  cc_log_argv("Executing compiler check command ", argv);
 
 #ifdef _WIN32
   PROCESS_INFORMATION pi;
   memset(&pi, 0x00, sizeof(pi));
   STARTUPINFO si;
   memset(&si, 0x00, sizeof(si));
 
-  char* path = find_executable_in_path(args->argv[0], nullptr, getenv("PATH"));
+  char* path = find_executable_in_path(argv[0], nullptr, getenv("PATH"));
```

Comment:
Don't rely on to_argv where it's not needed.
```suggestion
  char* path = find_executable_in_path(args[0].c_str(), nullptr, getenv("PATH"));
```

Reply:
Done.